### PR TITLE
[Az.RecoveryServices.Backup] Added support for setting snapshot consistency type in backup policy 

### PR DESCRIPTION
**Pull request by hiaga**
## Description

* Added option to set snapshot consistency type in policy cmdlets for creating or updating enhanced AzureVM policies.
* Fixed an issue while setting soft delete vault property. 

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices/help/New-AzRecoveryServicesBackupProtectionPolicy.md [hunk 1]
```diff
@@ -228,6 +229,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
+{{ Fill ProgressAction Description }}
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/New-AzRecoveryServicesBackupProtectionPolicy.md:233]:
```suggestion
```

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md [hunk 2]
```diff
@@ -194,6 +197,21 @@ Accept pipeline input: True (ByValue)
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
+{{ Fill ProgressAction Description }}
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md:201]:
```suggestion
```

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md [hunk 3]
```diff
@@ -154,6 +154,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
+{{ Fill ProgressAction Description }}
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md:158]:
```suggestion
```

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please fix the failed test cases

File: src/RecoveryServices/RecoveryServices/help/New-AzRecoveryServicesBackupProtectionPolicy.md [hunk 4]
```diff
@@ -18,8 +18,9 @@ New-AzRecoveryServicesBackupProtectionPolicy [-Name] <String> [-WorkloadType] <W
  [[-BackupManagementType] <BackupManagementType>] [[-RetentionPolicy] <RetentionPolicyBase>]
  [[-SchedulePolicy] <SchedulePolicyBase>] [[-MoveToArchiveTier] <Boolean>] [[-TieringMode] <TieringMode>]
  [[-TierAfterDuration] <Int32>] [[-TierAfterDurationType] <String>] [-BackupSnapshotResourceGroup <String>]
- [-BackupSnapshotResourceGroupSuffix <String>] [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>]
- [-WhatIf] [-Confirm] [<CommonParameters>]
+ [-BackupSnapshotResourceGroupSuffix <String>] [-SnapshotConsistencyType <SnapshotConsistencyType>]
+ [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>] [-ProgressAction <ActionPreference>] [-WhatIf]
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/New-AzRecoveryServicesBackupProtectionPolicy.md:22]:
```suggestion
 [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
```

File: src/RecoveryServices/RecoveryServices/help/New-AzRecoveryServicesBackupProtectionPolicy.md [hunk 5]
```diff
@@ -228,6 +229,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/New-AzRecoveryServicesBackupProtectionPolicy.md:232]: Remove ProgressAction

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md [hunk 6]
```diff
@@ -18,15 +18,18 @@ Modifies a Backup protection policy.
 Set-AzRecoveryServicesBackupProtectionPolicy [-Policy] <PolicyBase> [[-RetentionPolicy] <RetentionPolicyBase>]
  [[-SchedulePolicy] <SchedulePolicyBase>] [-MoveToArchiveTier <Boolean>] [-TieringMode <TieringMode>]
  [-TierAfterDuration <Int32>] [-TierAfterDurationType <String>] [-BackupSnapshotResourceGroup <String>]
- [-BackupSnapshotResourceGroupSuffix <String>] [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>]
- [-Token <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
+ [-BackupSnapshotResourceGroupSuffix <String>] [-SnapshotConsistencyType <SnapshotConsistencyType>]
+ [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>] [-Token <String>]
+ [-ProgressAction <ActionPreference>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md:23]:
```suggestion
 [-WhatIf] [-Confirm] [<CommonParameters>]
```

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md [hunk 7]
```diff
@@ -194,6 +197,21 @@ Accept pipeline input: True (ByValue)
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md:200]: Remove ProgressAction

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md [hunk 8]
```diff
@@ -17,15 +17,15 @@ Updates properties of a Vault.
 ```
 Set-AzRecoveryServicesVaultProperty [-SoftDeleteFeatureState <String>]
  [-DisableHybridBackupSecurityFeature <Boolean>] [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>]
- [-Token <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
+ [-Token <String>] [-ProgressAction <ActionPreference>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md:20]:
```suggestion
 [-Token <String>] [-WhatIf] [-Confirm] [<CommonParameters>]
```

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md [hunk 9]
```diff
@@ -17,15 +17,15 @@ Updates properties of a Vault.
...
 ```
 
 ### AzureRSVaultCMKParameterSet
 ```
 Set-AzRecoveryServicesVaultProperty [-VaultId <String>] [-DefaultProfile <IAzureContextContainer>]
  -EncryptionKeyId <String> [-KeyVaultSubscriptionId <String>] [-InfrastructureEncryption]
- [-UseSystemAssignedIdentity <Boolean>] [-UserAssignedIdentity <String>] [-WhatIf] [-Confirm]
- [<CommonParameters>]
+ [-UseSystemAssignedIdentity <Boolean>] [-UserAssignedIdentity <String>] [-ProgressAction <ActionPreference>]
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md:27]:
```suggestion
 [-UseSystemAssignedIdentity <Boolean>] [-UserAssignedIdentity <String>]
```

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md [hunk 10]
```diff
@@ -154,6 +154,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -ProgressAction
```
Comment (msJinLei) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesVaultProperty.md:157]: Remove ProgressAction

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by hiaga** [comment]
updated as per review comments

**Review by msJinLei — APPROVED** [review]

**Pull request merged**